File: java/Visitor/src/Order.java

import java.util.ArrayList;

public class Order implements Visitable {

	String name;

	private ArrayList<String> items = new ArrayList<String>();

	public Order(String name) {
		this.name = name;
	}

	public String getName() {
		return name;
	}

	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

	public void addItem(String item) {
		items.add(item);
	}

}

File: ruby/Visitor/order.rb
require_relative 'visitable'
require_relative 'visitor'

class Order < Visitable
  attr_accessor :name
  def initialize(name)
    @name = name
    @items = Array.new
  end

  def accept(visitor)
    visitor.visit(self)
  end

  def addItem(item)
    @items.insert(item)
  end
end